debugtools/DDR_VM/src/com/ibm/j9ddr/vm29/pointer/helper/J9ObjectHelper.java

```diff
@@ -68,27 +89,44 @@ public static boolean isIndexable(J9ObjectPointer objPointer) throws CorruptData
 	public static U32 flags(J9ObjectPointer objPointer) throws CorruptDataException
 	{		
 		if(J9BuildFlags.interp_flagsInClassSlot) {
-			long bitmask = (J9Consts.J9_REQUIRED_CLASS_ALIGNMENT - 1);
-			J9ClassPointer clazz = objPointer.clazz();
+			long bitmask = J9Consts.J9_REQUIRED_CLASS_ALIGNMENT - 1;
+			UDATA clazz = rawClazz(objPointer);
 			
-			return new U32(UDATA.cast(clazz).bitAnd(bitmask));
+			return new U32(clazz.bitAnd(bitmask));
 		} else {
 			throw new UnsupportedOperationException("Only builds with interp_flagsInClassSlot currently supported.");
 		}
 	}
 
+	public static long headerSize()
+	{
+		if (mixedMode) {
+			if (compressObjectReferences) {
+				return J9ObjectCompressed.SIZEOF;
+			}
+			return J9ObjectFull.SIZEOF;
+		}
+		return J9Object.SIZEOF;
+	}
+
+	public static UDATA rawClazz(J9ObjectPointer objPointer) throws CorruptDataException
+	{
+		if (mixedMode) {
+			if (compressObjectReferences) {
+				return J9ObjectCompressedPointer.cast(objPointer).clazz();
+			}
+			return J9ObjectFullPointer.cast(objPointer).clazz();
+		}
+		return UDATA.cast(objPointer.clazz());
+	}
+
 	public static J9ClassPointer clazz(J9ObjectPointer objPointer) throws CorruptDataException
 	{
 		J9ClassPointer classPointer = checkClassCache(objPointer);
 		if(null == classPointer) {
-			if(J9BuildFlags.interp_flagsInClassSlot) {
```

Comment:
Is `interp_flagsInClassSlot` another of those flags that can't be turned off?

Reply:
Yes - I'm not fixing anything to do with this in this PR.

Reply:
That's fine with me; I just wanted to be sure that the removal of the test of `interp_flagsInClassSlot` was intended and will be safe.